packages/x-date-pickers/src/tests/describeGregorianAdapter/testCalculations.ts

```diff
@@ -1,275 +1,529 @@
+    expect(adapter.getSeconds(testDateIso)).to.equal(0);
   });
 
-  it('Method: setMinutes', () => {
-    const updatedTime = adapter.formatByString(adapter.setMinutes(testDate, 12), formatDateTime);
-    expect(updatedTime).to.equal('2018-10-30 11:12:00');
+  it('Method: setYear', () => {
+    expect(adapter.setYear(testDateIso, 2011)).toEqualDateTime('2011-10-30T11:44:00.000Z');
   });
 
-  it('Method: setYear', () => {
-    const updatedTime = adapter.formatByString(adapter.setYear(testDate, 2011), formatDateTime);
-    expect(updatedTime).to.equal('2011-10-30 11:44:00');
+  it('Method: setMonth', () => {
+    expect(adapter.setMonth(testDateIso, 4)).toEqualDateTime('2018-05-30T11:44:00.000Z');
   });
 
   it('Method: setDate', () => {
-    const updatedTime = adapter.formatByString(adapter.setDate(testDate, 15), formatDateTime);
-    expect(updatedTime).to.equal('2018-10-15 11:44:00');
+    expect(adapter.setDate(testDateIso, 15)).toEqualDateTime('2018-10-15T11:44:00.000Z');
   });
 
-  it('Method: setSeconds', () => {
-    const updatedValue = adapter.formatByString(adapter.setSeconds(testDate, 11), formatDateTime);
+  it('Method: setHours', () => {
+    expect(adapter.setHours(testDateIso, 0)).toEqualDateTime('2018-10-30T00:44:00.000Z');
+  });
 
-    expect(updatedValue).to.equal('2018-10-30 11:44:11');
+  it('Method: setMinutes', () => {
+    expect(adapter.setMinutes(testDateIso, 12)).toEqualDateTime('2018-10-30T11:12:00.000Z');
   });
 
-  it('Method: isAfter', () => {
-    expect(adapter.isAfter(adapter.date()!, testDate)).to.equal(true);
-    expect(adapter.isAfter(testDate, adapter.date()!)).to.equal(false);
+  it('Method: setSeconds', () => {
+    expect(adapter.setSeconds(testDateIso, 11)).toEqualDateTime('2018-10-30T11:44:11.000Z');
   });
 
-  it('Method: isBefore', () => {
-    expect(adapter.isBefore(testDate, adapter.date()!)).to.equal(true);
-    expect(adapter.isBefore(adapter.date()!, testDate)).to.equal(false);
+  it('Method: getDaysInMonth', () => {
+    expect(adapter.getDaysInMonth(testDateIso)).to.equal(31);
+    expect(adapter.getDaysInMonth(testDateLocale)).to.equal(31);
+    expect(adapter.getDaysInMonth(adapter.addMonths(testDateIso, 1))).to.equal(30);
   });
 
-  it('Method: isAfterDay', () => {
-    const nextDay = adapter.addDays(testDate, 1);
+  it('Method: getNextMonth', () => {
+    expect(adapter.getNextMonth(testDateIso)).toEqualDateTime('2018-11-30T11:44:00.000Z');
+  });
 
-    expect(adapter.isAfterDay(nextDay, testDate)).to.equal(true);
-    expect(adapter.isAfterDay(testDate, nextDay)).to.equal(false);
+  it('Method: getPreviousMonth', () => {
+    expect(adapter.getPreviousMonth(testDateIso)).toEqualDateTime('2018-09-30T11:44:00.000Z');
   });
 
-  it('Method: isBeforeDay', () => {
-    const previousDay = adapter.addDays(testDate, -1);
+  it('Method: getMonthArray', () => {
+    const monthArray = adapter.getMonthArray(testDateIso);
+    let expectedDate = adapter.date('2018-01-01 00:00:00')!;
```

Comment:
Nitpick: it seems that everywhere else the date-time string includes the separating `T` character. 🤔 
```suggestion
    let expectedDate = adapter.date('2018-01-01T00:00:00')!;
```